src/core/data/field/field.h

```diff
@@ -71,6 +72,21 @@ namespace core
         std::string name_{"No Name"};
         PhysicalQuantity qty_;
     };
+
+
+
+
+    template<typename Field>
+    void average(Field const& f1, Field const& f2, Field& avg)
```

Comment:
could cause conflicts if another templated "average" function is ever made. (with three parameters)
perhaps as a static function on the class?

Reply:
or alternatively something like 
```
    template<typename T>
    void average(Field<T> const& f1, Field<T> const& f2, Field<T>& avg)
```

would block conflicts